runtime/plugins/comment/comment.lua

```diff
@@ -61,17 +61,15 @@ ft["zig"] = "// %s"
 ft["zscript"] = "// %s"
 ft["zsh"] = "# %s"
 
-local last_ft
-
 function updateCommentType(buf)
-    if buf.Settings["commenttype"] == nil or (last_ft ~= buf.Settings["filetype"] and last_ft ~= nil) then
+    -- NOTE: Don't use SetOptionNative() to set "comment.type",
+    -- otherwise "comment.type" can't be reset by a "filetype" change.
```

Comment:
It is not nearly obvious *why* can't it be reset by a "filetype" change if we use `SetOptionNative()` (it took me some time to figure it out, for instance). We should explain it better, e.g. "Don't use SetOptionNative() to set "comment.type", otherwise "comment.type" will be marked as locally overridden and will not be updated when "filetype" changes."

Or actually it seems better to:
1. use `DoSetOptionNative()` instead of setting the option directly
2. add documentation to both `SetOptionNative()` and `DoSetOptionNative()`, so that everyone knows what is the difference between them and when to use each of them, and then we don't need this comment here.

...Another option would be to somehow extend `RegisterCommonOption()` to allow registering default per-filetype values of an option, not just its default global value, at init time, - so that `updateCommentType()` would not be needed at all.

Reply:
@dmaluka 

`DoSetOptionNative()` seems like an internal function to me, the name of it is also quite confusing against `SetOptionNative()` as well.

Would adding a function like `SetOptionPersistent(bool persistent)` which sets `b.LocalSettings[option]` be enough for now?

Reply:
As matter of fact, the line between internal and external functions here is pretty blurred. (Technically `DoSetOptionNative()` is external, it is already exported to plugins, although not on purpose but as a side effect of exporting it to other modules *inside* micro, which is a consequence of the bizarre design of micro's plugin system.)

> Would adding a function like `SetOptionPersistence(bool persistent)` which sets `b.LocalSettings[option]`

Maybe... @JoeKar what do you think?

Reply:
> It is not nearly obvious _why_ can't it be reset by a "filetype" change if we use `SetOptionNative()` (it took me some time to figure it out, for instance).

I thought you can remember, after the long review road of #3343. :wink: 

> > Would adding a function like `SetOptionPersistence(bool persistent)` which sets `b.LocalSettings[option]`
>
> Maybe... @JoeKar what do you think?

So `SetOptionNative()` calls then `SetOptionNativeMark(option string, nativeValue interface{}, mark(Local) bool)`.
The same then for consistent reason for `SetGlobalOptionNative()` -> `SetGlobalOptionNativeMark(option string, nativeValue interface{}, mark(Modified) bool)`

The word `persistent` resp. `persistence` smells a bit of writing it persistent into the users configuration.

BTW: Wasn't the extension with a further parameter of these functions temporary part of #3343, but rejected due to the fact a of the introduction of a further parameter?

Anyway, if fine with that adjustment, in case it helps to improve the code/interfaces. Indeed we can then drop the introduced comment in the `comment` plugin.

Reply:
I think what @Neko-Box-Coder meant is a separate function just for toggling the option's persistence state, independently of setting the option value.

Reply:
Hm, wouldn't it over-complicate the situation (complex solution for simple problem)?
Where does `SetOptionNative()` and `DoSetOptionNative()` (current naming) not fulfill the needs to add this additional function to control the marking alone?

Reply:
I have the same thoughts.

Reply:
Okay, that's fine. 

It's just I am not sure how to rename `DoSetOptionNative()` such that it is not confusing and doesn't require a comment explaining what the difference is.

Would it work if I add a proxy function called something like `SetNonReloadableOptionNative()` (or some other names) which does the same as `SetOptionNative()` but without setting the `LocalSettings` field.

Reply:
It's fine to add documentation I think but it would be **difficult to rename or change** the function signature if we decide to use it in the plugin (which other people will do the same as well if needed)

I still stand by what I said before:

> It's just I am not sure how to rename `DoSetOptionNative()` such that it is not confusing and doesn't require a comment explaining what the difference is.
> 
> Would it work if I add a proxy function called something like `SetNonReloadableOptionNative()` (or some other names) which does the same as `SetOptionNative()` but without setting the `LocalSettings` field.

Or even a step further where we make `DoSetOptionNative()` as private just like `doSetGlobalOptionNative()`. The only place it is using `DoSetOptionNative()` is `internal/action/command.go:608` after all.

Reply:
I don't think `DoSetOptionNative` is a particularly bad name (but you are welcome to suggest a better one). It aligns with what it does (and what is described in the documentation I've suggested above): just sets the option for this buffer and does nothing more.

Also I hope it is unlikely that any other plugin will actually want to use it. The `comment` plugin's use case is unusual in this regard.

> Or even a step further where we make `DoSetOptionNative()` as private just like `doSetGlobalOptionNative()`. The only place it is using `DoSetOptionNative()` is `internal/action/command.go:608` after all.

`internal/action/command.go` is in a different package. Which is why we made `DoSetOptionNative()` public in the first place. Try making it private and compiling micro.